frameworks/kafka/tests/test_check_topic_list_on_zk_restart.py

```diff
@@ -0,0 +1,179 @@
+    finally:
+        sdk_install.uninstall(
+            config.ZOOKEEPER_PACKAGE_NAME,
+            config.ZOOKEEPER_SERVICE_NAME,
+        )
+        if sdk_utils.is_strict_mode():
+            sdk_security.delete_service_account(
+                service_account_name=zk_account,
+                service_account_secret=zk_secret,
+            )
+
+
+@pytest.fixture(scope="module", autouse=True)
+def kafka_client():
+    try:
+        kafka_client = client.KafkaClient(
+            "kafka-client",
+            config.PACKAGE_NAME,
+            config.SERVICE_NAME,
+        )
+        kafka_client.install()
+
+        yield kafka_client
+    finally:
+        kafka_client.uninstall()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def kafka_server(zookeeper_server):
+    try:
+
+        # Get the zookeeper DNS values
+        zookeeper_dns = sdk_cmd.svc_cli(
+            zookeeper_server["package_name"],
+            zookeeper_server["service"]["name"],
+            "endpoint clientport",
+            parse_json=True,
+        )[1]["dns"]
+
+        sdk_install.uninstall(
+            config.PACKAGE_NAME, config.SERVICE_NAME
+        )
+
+        config.install(
+            config.PACKAGE_NAME,
+            config.SERVICE_NAME,
+            config.DEFAULT_BROKER_COUNT,
+            additional_options={
+                "kafka": {
+                    "kafka_zookeeper_uri": ",".join(
+                        zookeeper_dns
+                    )
+                }
+            },
+        )
+
+        # wait for brokers to finish registering before starting tests
+        test_utils.broker_count_check(
+            config.DEFAULT_BROKER_COUNT,
+            service_name=config.SERVICE_NAME,
+        )
+
+        yield {
+            "package_name": config.PACKAGE_NAME,
+            "service": {"name": config.SERVICE_NAME},
+        }
+    finally:
+        sdk_install.uninstall(
+            config.PACKAGE_NAME, config.SERVICE_NAME
+        )
+
+
+def topic_create(kafka_server: dict):
+    return test_utils.create_topic(
+        config.EPHEMERAL_TOPIC_NAME,
+        kafka_server["service"]["name"],
+    )
+
+
+def fetch_topic(kafka_server: dict):
+    _, topic_list, _ = sdk_cmd.svc_cli(
+        config.PACKAGE_NAME,
+        kafka_server["service"]["name"],
+        "topic list",
+        parse_json=True,
+    )
+    return topic_list
+
+
+def restart_zookeeper_node(id: int):
+    sdk_cmd.svc_cli(
+        config.ZOOKEEPER_PACKAGE_NAME,
+        config.ZOOKEEPER_SERVICE_NAME,
+        "pod restart zookeeper-{}".format(id),
+    )
+    sdk_plan.wait_for_kicked_off_recovery(
+        config.ZOOKEEPER_SERVICE_NAME
+    )
+    sdk_plan.wait_for_completed_recovery(
+        config.ZOOKEEPER_SERVICE_NAME
+    )
+
+
+@pytest.mark.sanity
+@pytest.mark.zookeeper
+def test_check_topic_list_on_zk_restart(kafka_server: dict):
+    ID = 0
+    topic_create(kafka_server)
+    topic_list_before = fetch_topic(kafka_server)
+    restart_zookeeper_node(ID)
```

Comment:
You need to restart all the nodes to ensure re connect actually works right? because you are not specifying the number of zk nodes so it by default launches 3 nodes.

Reply:
As discussed, I am considering only one Zk node for the the process. the ZK_TASK_COUNT is set to 2. Which means that I'll be waiting for at-least 1 server task(other one is metrics task).

Reply:
Yeah the sdk_install.install will wait for 2 task before proceeding to the next step but the framework will have 3x2 tasks as default is 3 zk nodes

Reply:
I omitted the {"node": {"count": 1}} because it is not allowed as the min value of zk nodes should be 3. However, if we restart all the 3 zk nodes, then also it verifies the purpose. I've tested this locally as well. Since node count cannot be < 3, we can consider restarting of all the three nodes.
cc: @zmalik

Reply:
can you change the test to restart all of them and wait for them to be up?

Reply:
Done. We are restarting all the Zk nodes now.